src/main/java/net/imagej/ops/filter/bilateral/DefaultBilateral.java

```diff
@@ -0,0 +1,148 @@
+import net.imglib2.algorithm.neighborhood.RectangleNeighborhood;
+import net.imglib2.algorithm.neighborhood.RectangleNeighborhoodFactory;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.view.Views;
+
+import org.scijava.Priority;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+/**
+ * Performs a bilateral filter on an image. The sigmaS parameter refers to the
+ * spatial smoothing parameter; the greater the sigma, the smoother the image.
+ * The sigmaR parameter refers to the range smoothing parameter. The greater the
+ * sigma, the greater the effect of intensity differences. The radius parameter
+ * refers to the square that is considered when doing the filter on each
+ * individual picture.
+ *
+ * @author Gabe Selzer
+ * @param <I>
+ * @param <O>
+ */
+
+@Plugin(type = Ops.Filter.Bilateral.class, priority = Priority.NORMAL_PRIORITY)
+public class DefaultBilateral<I extends RealType<I>, O extends RealType<O>>
+	extends
+	AbstractUnaryComputerOp<RandomAccessibleInterval<I>, RandomAccessibleInterval<O>>
+	implements Ops.Filter.Bilateral
+{
+
+	public final static int MIN_DIMS = 2;
+
+	public final static int MAX_DIMS = 2;
+
+	@Parameter
+	private double sigmaR;
+
+	@Parameter
+	private double sigmaS;
+
+	@Parameter
+	private int radius;
+
+	private static double gauss(final double x, final double sigma) {
+		final double mu = 0.0;
+		return (1 / (sigma * Math.sqrt(2 * Math.PI))) * Math.exp((-0.5 * (x - mu) *
+			(x - mu)) / (sigma * sigma));
+	}
+
+	@Override
+	public void compute(final RandomAccessibleInterval<I> input,
+		final RandomAccessibleInterval<O> output)
+	{
+		if (input.numDimensions() != 2) {
+			throw new IllegalArgumentException("Input must be two dimensional");
+		}
+
+		final long[] size = new long[input.numDimensions()];
+		input.dimensions(size);
+
+		final RandomAccess<O> cr = output.randomAccess();
+		final Cursor<I> cp = Views.iterable(input).localizingCursor();
+		final long[] p = new long[input.numDimensions()];
+		final int[] q = new int[input.numDimensions()];
+		final long[] mi = new long[input.numDimensions()];
+		final long[] ma = new long[input.numDimensions()];
+		final long mma1 = input.max(0);
+		final long mma2 = input.max(1);
+		Neighborhood<I> rn;
+		Cursor<I> cq;
+		while (cp.hasNext()) {
+			cp.fwd();
+			cp.localize(p);
+			double d;
+			cp.localize(mi);
+			cp.localize(ma);
+			mi[0] = Math.max(0, mi[0] - radius);
+			mi[1] = Math.max(0, mi[1] - radius);
+			ma[0] = Math.min(mma1, ma[0] + radius);
+			ma[1] = Math.min(mma2, ma[1] + radius);
+			final Interval in = new FinalInterval(mi, ma);
+			final RectangleNeighborhoodFactory<I> fac = RectangleNeighborhood
+				.factory();
+			rn = fac.create(p, mi, ma, in, input.randomAccess());
+			cq = rn.localizingCursor();
+			double s, v = 0.0;
+			double w = 0.0;
+			do {
+				cq.fwd();
+				cq.localize(q);
+				d = ((p[0] - q[0] - mi[0]) * (p[0] - q[0] - mi[0])) + ((p[1] - q[1] -
```

Comment:
You could refactor this into distance function that also works with _n_-dimensional inputs